tests/Integration/TestCase.php

```diff
@@ -168,6 +177,81 @@ public function create_test_post( string $post_status = 'publish' ): int {
 		return self::factory()->post->create( $post_data );
 	}
 
+	/**
+	 * Create test posts in sequence.
+	 *
+	 * @param int    $num_of_posts Optional. Number of posts we need to create.
+	 * @param string $post_type Optional. Type of the posts.
+	 * @param string $post_status Optional. Status of the posts.
+	 *
+	 * @return WP_Post[]
+	 */
+	public function create_and_get_test_posts( int $num_of_posts = 1, $post_type = 'post', $post_status = 'publish' ) {
+		$post_ids = $this->create_posts_and_get_ids( $num_of_posts, $post_type, $post_status );
+
+		return $this->get_test_posts( $post_ids );
+	}
+
+	/**
+	 * Create test posts in sequence.
+	 *
+	 * @param int    $num_of_posts Optional. Number of posts we need to create.
+	 * @param string $post_type Optional. Type of the posts.
+	 * @param string $post_status Optional. Status of the posts.
+	 *
+	 * @return int[]
+	 */
+	private function create_posts_and_get_ids( int $num_of_posts = 1, $post_type = 'post', $post_status = 'publish' ) {
+		/**
+		 * Variable.
+		 *
+		 * @var int[]
+		 */
+		$post_ids = array();
+		$date     = new DateTime( '2009-12-31', new DateTimeZone( 'America/New_York' ) ); // Start date with timezone to replicate scenarios like real world.
+
+		for ( $i = 1; $i <= $num_of_posts; $i++ ) {
+			$post_date = date_add( $date, date_interval_create_from_date_string( '1 days' ) ); // Increment by 1 day like sequence.
+			$post_id   = self::factory()->post->create(
+				array(
+					'post_type'     => $post_type,
+					'post_status'   => $post_status,
+					'post_title'    => "Title $i-($post_status)",
+					'post_author'   => $i,
+					'post_content'  => "Content $i",
+					'post_date'     => $post_date->format( WP_DATE_TIME_FORMAT ),
+					'post_date_gmt' => gmdate( WP_DATE_TIME_FORMAT, $post_date->getTimestamp() ),
+				)
+			);
+
+			array_push( $post_ids, $post_id );
+		}
+
+		return $post_ids;
+	}
+
+	/**
+	 * Get given test posts.
+	 *
+	 * @param int[] $post_ids IDs of the posts.
+	 *
+	 * @return WP_Post[]
+	 */
+	private function get_test_posts( $post_ids = array() ) {
+		$posts = array();
+
+		foreach ( $post_ids as $post_id ) {
+			array_push( $posts, get_post( $post_id ) );
```

Comment:
Was not able to get draft posts using `get_posts` function so chose this approach.